### PR TITLE
New OML and ANML configs and  metacl experiment class enhancements

### PR DESCRIPTION
**Pull request by mvacaporale**
This PR include new OML and ANML experiments, mostly ablation studies, and updates the the MetaContinualLearningExperiment class.

For the experiment class
* We can optionally specify to use first order or second order gradients, for ablations
* We can override the a new function sample_slow_data which helps to sample i.i.d when desired
* We can optionally skip the lr sweep / grid-search (in associated OML mixin)

For the experiments
* There are configs for training with first order derivations with OML and ANML
* There are configs for training i.i.d. with OML and ANML
* There's a config training OML with kwinners

### CODE REVIEW

File: nupic/research/frameworks/vernon/experiments/meta_cl_experiment.py [hunk 1]
```diff
@@ -203,6 +207,8 @@ def compute_class_indices(cls, config, dataset, mode="all", sample_size=None):
         for idx, (_, target) in enumerate(dataset):
             if isinstance(target, torch.Tensor):
                 target = target.item()
+            elif isinstance(target, np.integer):
+                target = int(target)
```
Comment (lucasosouza) [inline nupic/research/frameworks/vernon/experiments/meta_cl_experiment.py:211]: Just curious, any specific cases where this requirement came up? I would expect datasets loaded by pytorch to use tensor format.

File: nupic/research/frameworks/vernon/mixins/oml.py [hunk 2]
```diff
@@ -309,7 +320,11 @@ def run_meta_testing_phase(self, num_classes_learned):
         memorize without forgetting.
         """
 
-        lr = self.find_best_lr(num_classes_learned)
+        # Decide on the lr to use.
+        if self.run_lr_sweep:
+            lr = self.find_best_lr(num_classes_learned)
+        else:
+            lr = self.lr_sweep_range[0]
```
Comment (lucasosouza) [inline nupic/research/frameworks/vernon/mixins/oml.py:327]: This defaults to 1e-1. The most common LR we've seen in the experiments, found through grid search in test time, is 1e-3. I think we should default to that.

File: projects/meta_cl/networks/oml_networks.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+# ----------------------------------------------------------------------
+
+import torch.nn as nn
+
+from nupic.torch.modules.k_winners import KWinners
+
+
+class KWinnerOMLNetwork(nn.Module):
+    """
+    This reproduces the network from the `OML`_ repository.
```
Comment (lucasosouza) [inline projects/meta_cl/networks/oml_networks.py:29]: Docstring say it reproduces network, we should add there is a twist.

File: projects/meta_cl/experiment_classes.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+from nupic.research.frameworks.vernon import MetaContinualLearningExperiment, mixins
+
+
+class OMLExperiment(mixins.OnlineMetaLearning,
+                    MetaContinualLearningExperiment):
+    pass
+
+
+class IIDTrainingOMLExperiment(OMLExperiment):
+    def sample_slow_data(self, tasks):
```
Comment (lucasosouza) [inline projects/meta_cl/experiment_classes.py:31]: Isn't this just removing the replay set, instead of replacing for one that is sampled from all existing classes? Is that a fair comparison?

You also argued the difference for the iid setting is that in the inner loop we are no longer seeing 20 images from a single class, but 20 images sampled from any class. Is that change also part of this PR?

**Review by lucasosouza — APPROVED** [review]
Left some questions, otherwise looks good.

File: nupic/research/frameworks/vernon/experiments/meta_cl_experiment.py (hunk 1 above)
Comment (mvacaporale) [reply nupic/research/frameworks/vernon/experiments/meta_cl_experiment.py:211]: Our GSC preprocessed dataset used numpy.int64 types, not sure why. It seems it's possible for the targets to be int, torch.int, or numpy.int

File: nupic/research/frameworks/vernon/mixins/oml.py (hunk 2 above)
Comment (mvacaporale) [reply nupic/research/frameworks/vernon/mixins/oml.py:327]: Good point, I'll make that change.

File: projects/meta_cl/experiment_classes.py (hunk 4 above)
Comment (mvacaporale) [reply projects/meta_cl/experiment_classes.py:31]: Per each question:
1) The replay set is still sampled. In fact, it's the only data that would now get used in the outer loop. For instance, if there were 30 images before, 15 from the replay set, and 15 sampled from the task(s), then now there would be only 15 from the replay set.

2) The inner loop was already capable of this. You do this by setting `tasks_per_epoch=20` and `num_fast_steps=1`. The tasks are sampled randomly and you take `num_fast_steps` per task. Thus, you take one gradient step over one image sampled accross 20 random tasks, as you request.
